modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -412,6 +412,14 @@ private Column<?, Object> _getColumn(
 			entityField.getFilterableName(null));
 	}
 
+	private String _getDateTimePattern(String value) {
+		if (value.length() > 20) {
+			return "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+		}
+
+		return "yyyy-MM-dd'T'HH:mm:ss";
+	}
+
```

Comment:
@p-albuquerque please check if we can use this [method](https://github.com/liferay/liferay-portal/blob/1206124b6f7b7d569ceaf75a45f8dffac6ab062f/modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java#L175). Thanks!

Reply:
It works!